### PR TITLE
[UT] increase coverage ut for kubeadm/app/features

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
[UT] increase coverage ut for kubeadm/app/features

before:

```
        k8s.io/kubernetes/cmd/kubeadm/app/features      coverage: 79.7% of statements
```

after this change:

```
        k8s.io/kubernetes/cmd/kubeadm/app/features      coverage: 94.9% of statements
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by Tusenka — APPROVED** [review]

File: cmd/kubeadm/app/features/features_test.go [hunk 1]
```diff
@@ -207,6 +208,11 @@ func TestCheckDeprecatedFlags(t *testing.T) {
 			features:    map[string]bool{"feature1": true},
 			expectedMsg: map[string]string{},
 		},
+		{
+			name:        "invalid feature",
+			features:    map[string]bool{"": true},
```
Comment (pacoxu) [inline cmd/kubeadm/app/features/features_test.go:213]: empty FG is a special one, and how about add a general invalid FG like 'feature2' or 'invalid'.

File: cmd/kubeadm/app/features/features_test.go [hunk 2]
```diff
@@ -218,3 +224,53 @@ func TestCheckDeprecatedFlags(t *testing.T) {
...
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := Supports(InitFeatureGates, test.featureName); got != test.want {
+				t.Errorf("Supports() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestKeys(t *testing.T) {
```
Comment (pacoxu) [inline cmd/kubeadm/app/features/features_test.go:254]: `Keys()` is not used anymore. We may just clean it up.

**Review by pacoxu — COMMENTED** [review]
BTW, there is a discussion about refactoring the code here in https://github.com/kubernetes/kubernetes/pull/118270#issuecomment-1564318620.
/assign @SataQiu

File: cmd/kubeadm/app/features/features.go [hunk 3]
```diff
@@ -102,15 +102,6 @@ func Supports(featureList FeatureList, featureName string) bool {
 	return false
 }
 
-// Keys returns a slice of feature names for a given feature set
-func Keys(featureList FeatureList) []string {
```
Comment (chendave) [inline cmd/kubeadm/app/features/features.go:106]: removing this seems like should be in another commit instead of here, fine as is.

File: cmd/kubeadm/app/features/features_test.go [hunk 4]
```diff
@@ -207,13 +207,44 @@ func TestCheckDeprecatedFlags(t *testing.T) {
...
+		{
+			name:        "the feature is supported",
+			featureName: PublicKeysECDSA,
+			want:        true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := Supports(InitFeatureGates, test.featureName); got != test.want {
+				t.Errorf("Supports() = %v, want %v", got, test.want)
```
Comment (chendave) [inline cmd/kubeadm/app/features/features_test.go:247]: we'd better not print the err msg with this style next time, but instead should similar like this "export Supports return: %v, but got: %v ..." to align with others in this component.

**Review by chendave — COMMENTED** [review]
/lgtm
